addons/account/models/account_move.py

```diff
@@ -5343,70 +5354,57 @@ def _add_base_line_caba_exchange_diff_line_vals(existing_line_vals_list, values,
+            currency_rate = currency_line.balance / currency_line.amount_currency if currency_line.amount_currency else 1.0
 
-                    if move.company_currency_id.is_zero(balance):
-                        continue
+            for grouping_key, values in account_vals_to_fix.items():
 
-                    if values.get('tax_repartition_line_id'):
-                        # Tax line.
-                        _add_tax_line_caba_exchange_diff_line_vals(
-                            move_vals['line_ids'],
-                            values,
-                            caba_rounding_diff_label,
-                            balance,
-                            values['amount_currency'],
-                        )
-                    else:
-                        # Base line.
-                        _add_base_line_caba_exchange_diff_line_vals(
-                            move_vals['line_ids'],
-                            values,
-                            caba_rounding_diff_label,
-                            balance,
-                            values['amount_currency'],
-                        )
-
-                elif not self.env['res.currency'].browse(values['currency_id']).is_zero(values['amount_currency']):
-                    # Then, there is a rounding error due to multiple payments on the foreign currency amount
+                if currency.is_zero(values['amount_currency']):
+                    continue
 
-                    # To know which rate to use for the adjustment, get the rate used by the most recent cash basis move
-                    caba_adjustment_currency_rates = {}
-                    currency_line = cash_basis_moves[-1].line_ids.filtered(lambda x: x.currency_id.id == values['currency_id'])[0]
-                    caba_adjustment_currency_rates[currency_line.currency_id.id] = currency_line.balance / currency_line.amount_currency
+                # There is a rounding error due to multiple payments on the foreign currency amount
+                balance = currency.round(currency_rate * values['amount_currency'])
+
+                if values.get('tax_repartition_line_id'):
+                    # Tax line
+                    _add_tax_line_caba_exchange_diff_line_vals(
+                        move_vals['line_ids'],
+                        values,
+                        caba_rounding_diff_label,
+                        balance,
+                        values['amount_currency'],
+                    )
+                    last_line_vals = move_vals['line_ids'][-1][2]
 
-                    balance = caba_adjustment_currency_rates[values['currency_id']] * values['amount_currency']
+                    if last_line_vals['account_id'] in result['caba_lines_to_reconcile'][move.id]:
+                        result['caba_lines_to_reconcile'][move.id][last_line_vals['account_id']]['sequence'] = last_line_vals['sequence']
```

Comment:
I don't get this. What does sequence have to do with all that ? This needs a least comment if it's necessary